Feature/scss layout setup

Jammanb0 opened the pull request:
## 개요
- 파일 구조를 피드가 기획본과 비교하여 세부적으로 수립하고, 각 페이지별 레이아웃을 설정하였습니다.
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

<!---- Resolves: #(Issue Number) -->

## PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [x] 주석 추가 및 수정
- [x] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [x] 파일 혹은 폴더명 수정
- [x] 파일 혹은 폴더 삭제

## 📸스크린샷 (선택)
![image](https://github.com/user-attachments/assets/74d35fff-e972-48dd-8926-ced97def9339)
![image](https://github.com/user-attachments/assets/8449b9a9-2773-4542-a98d-6c64abebdcf0)


## 💬 공유사항 to 리뷰어
- 피그마 기획본을 바탕으로 페이지를 상세하게 구분하였습니다.
  - [피그마 기획본 링크](https://www.figma.com/design/wchvqXNKP67z2UY0mC0Fym/Travel-product?node-id=245-487&p=f&t=DICuT2iMLZN0LMGc-0)
- 이에 따라 router.tsx에서 라우팅 재설정을 진행하였습니다.
- 피그마 기획본을 바탕으로 페이지별 동일한 구조를 가지는 페이지별로 묶어서 스타일링 하였습니다.
- 페이지 전용 layout 폴더를 만들었습니다.
- SCSS의 속성값들이 속성 그룹별로 정렬되도록  Stylelint 설정을 변경하였습니다.
- 윈도우와 맥 OS 간의 줄바꿈 관련 충돌이 없도록 수정했습니다.
- 레이아웃을 얼마나 어떻게 잡아야할지 감이 크게 잡히지 않아, content 부분의 큰 틀만 잡아둔 상태입니다. 추후 SCSS 클래스명 등을 자유롭게 변경하여 사용하시면 될 것 같습니당
- 라우팅 설정을 하며 임시로 엔드포인트(?)를 작성했는데 추후 논의가 필요할 것 같습니다.
- 페이지 레이아웃 설정을 하기 위해 router.tsx 부분의 requiredRole을 임시 주석 처리하였습니다.
- url 접속으로 페이지를 확인해보시면 pr과 함께 공유한 사진 처럼 파란 박스 안에 요소, 또는 요소들이 있는 것을 보실 수 있습니다.
  - 실제 콘텐츠가 들어갈 부분은 파란 박스 안의 요소 영역이며, 색깔 구분으로 쉽게 확인 가능하도록 임시로 배경색을 넣어두었음을 알려드립니다.
- pr 양식이 자동으로 생기지 않아 확인해보니, pr 양식이 main 브랜치에 없는 상황을 확인했습니다. 이에 pr 양식을 생성하여 main 브랜치에서 push 하였으니 확인 부탁드립니다.
- SCSS 처음 사용해봐서, 혹시나 잘못된 부분, 모자란 부분 있다면 언제든지 피드백 주세용
<!--- 리뷰어가 중점적으로 봐줬으면 좋겠는 부분이 있으면 적어주세요. -->
<!--- 논의해야할 부분이 있다면 적어주세요.-->
<!--- ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요? -->

## PR Checklist

PR이 다음 요구 사항을 충족하는지 확인하세요.

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] Eslint 및 Prettier 규칙을 준수했습니다.
- [x] origin/develop 브랜치에서의 최신 커밋을 확인하고 반영했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Jammanb0 committed:
chore: .gitkeep 파일 제거

Jammanb0 committed:
chore: husky pre-commit 규칙 수정

Jammanb0 committed:
chore: stylelintrc 규칙 수정

- no-empty-source 규칙을 비활성화

Jammanb0 committed:
feat: 공통 SCSS 파일 생성

Jammanb0 committed:
feat: 전역 SCSS 설정 파일 import

Jammanb0 committed:
feat: main.tsx CSS 파일 SCSS로 변경

Jammanb0 committed:
feat: reset.scss에 Eric Meyer's Reset CSS 적용

Jammanb0 committed:
feat: 상품 상세 페이지 추가

Jammanb0 committed:
chore: 윈도우, 맥 OS 줄바꿈 처리 방식 호환을 위한 lint 설정 추가

- 줄바꿈 처리를 각 운영체제에 맞도록 auto로 설정

Jammanb0 committed:
chore: pages/home 폴더 삭제

- 피그마 기획본을 참고하여 사용하지 않는 페이지 삭제

Jammanb0 committed:
chore: layout 폴더에서 commonLayout 폴더로 이름 변경

- 페이지 레이아웃 폴더와 구분하기 위한 목적으로 진행

Jammanb0 committed:
feat: 피그마 기획본에 따른 페이지 추가 생성 및 내용 수정

- 장바구니 페이지 생성
- 결제 진행 페이지 생성
- 결제 완료 페이지 생성
- 회원 정보 수정 페이지 생성
- 상품 상세 페이지 생성

Jammanb0 committed:
chore: 페이지 생성 및 삭제에 따른 router 수정

- 페이지 레이아웃 구성을 위해 임시로 requiredRole을 주석 처리함. 추후 수정 필요

Jammanb0 committed:
chore: 메인 페이지 loader.ts 및 scss 파일 생성

Jammanb0 committed:
chore: banner 폴더 구조 세분화

- 메인 페이지 배너와 상품 목록 페이지의 배너로 구분

Jammanb0 committed:
chore: husky 관련 deprecation 경고 수정

Jammanb0 committed:
feat: footer.tsx 및 scss 파일 생성

Jammanb0 committed:
feat: header 컴포넌트 Default/Admin 분리

Jammanb0 committed:
feat: 페이지 레이아웃 구조 생성

Jammanb0 committed:
feat: router에 페이지 layout 적용

Jammanb0 committed:
feat: core.scss 생성

Jammanb0 committed:
chore: 레이아웃 설정을 위한 requiredRole 임시 주석 처리

Jammanb0 committed:
chore: 루트에 위치한 App.css, index.css 제거

Jammanb0 committed:
chore: app.css 임포트 구문 삭제

Jammanb0 committed:
feat: 전역 단위 스타일링 설정 추가

Jammanb0 committed:
chore: 파일 구조 변경 및 scss 파일 추가

Jammanb0 committed:
feat: router에 추가한 레이아웃 폼 적용

Jammanb0 committed:
chore: stylelint, cspell 파일 수정

Jammanb0 committed:
style: stylelint 설정에 따른 css 속성 재정렬 진행

Jammanb0 committed:
feat: 줄간격 기본값 설정

Jammanb0 committed:
chore: styles 파일을 통한 전역 변수 및 mixns 사용

Jammanb0 committed:
feat: 마이(유저)페이지 배너 폴더와 유저 개인 정보 수정 레이아웃 폴더 생성

Jammanb0 committed:
fix: import 오류 수정 및 파일명 변경

Jammanb0 committed:
feat: layout 템플릿, 피그마 기획본에 맞춰 양식 설정

Jammanb0 committed:
chore: _mixins 파일, stylelint 설정 파일 수정

- stylelint 설정에서 flex-grow 속성 정렬 규칙 포함

Jammanb0 committed:
fix: router.tsx 파일 잘못된 라우팅 수정 및 주석 수정

- UserLayout 중복 라우팅 수정

Jammanb0 committed:
feat: 장바구니 페이지 레이아웃 구현

Jammanb0 committed:
feat: 메인 페이지 레이아웃 구현

Jammanb0 committed:
feat: 상품 목록 페이지 레이아웃 구현

Jammanb0 committed:
feat: 상품 상세 페이지 레이아웃 구현

Jammanb0 committed:
feat: 예약 페이지 레이아웃 구현

Jammanb0 committed:
feat: 결제 진행 페이지 레이아웃 구현

Jammanb0 committed:
feat: 결제 완료 페이지 레이아웃 구현

Jammanb0 committed:
feat: 마이페이지(유저 페이지) 레아이웃 구현

Jammanb0 committed:
feat: 회원 정보 수정페이지 레이아웃 구현

Jammanb0 committed:
chore: 팀에서 공통으로 사용하기로 결정된 이슈 및 pr 양식 적용

Jammanb0 committed:
fix: pr 양식 오타 수정

Shin-Yu-1 reviewed (approved):
고생 많으셨습니다! 👍

sjoon627 commented on src/layouts/DefaultLayout/DefaultLayout.module.scss:
```diff
@@ -0,0 +1,9 @@
+
+.baseForm {
+  width: core.$default-layout-width;
+  max-width: core.$default-layout-width;
```
```suggestion
  max-width: core.$default-layout-max-width;
```
오타인거 같은데 한번 확인부탁드려요

sjoon627 reviewed (approved):
너무 고생많으셨습니다!! 👍

vw로 너비를 지정하셨던데 반응형 웹일까요?
min-width나 반응형 스타일 관련해서 이야기가 필요할거 같아서 여쭤봅니담

Jammanb0 committed:
fix: 레아이웃의 DefaultLayout.module.scss 오타 수정

- max-width: core.$default-layout-width; -> max-width: core.$default-layout-max-width;
